File: .github/scripts/cleanup_firebase.py
import os
import requests
import json
def delete_all_releases():
    project_number = os.getenv('FIREBASE_PROJECT_NUMBER')
    app_id = os.getenv('FIREBASE_APP_ID')
    token = os.getenv('GOOGLE_ACCESS_TOKEN')
    base_url = "https://firebaseappdistribution.googleapis.com/v1"
    headers = {
        "Authorization": f"Bearer {token}",
        "Content-Type": "application/json"
    }
    list_url = f"{base_url}/projects/{project_number}/apps/{app_id}/releases"
    response = requests.get(list_url, headers=headers)
    if response.status_code != 200:
        return
    releases = response.json().get('releases', [])
    if not releases:
        return
    release_names = [r['name'] for r in releases]
    batch_delete_url = f"{base_url}/projects/{project_number}/apps/{app_id}/releases:batchDelete"
    payload = {
        "names": release_names
    }
    del_res = requests.post(batch_delete_url, headers=headers, data=json.dumps(payload))
    if del_res.status_code == 200:
        pass
    else:
        pass
if __name__ == "__main__":
    delete_all_releases()

File: .github/scripts/clean/cleanup_firebase.dart
import 'dart:io';
import 'dart:convert';
import 'package:dio/dio.dart';
import 'firebase_api_client.dart';
import 'firebase_cleanup_service.dart';

void main() async {
  final token = Platform.environment['GOOGLE_ACCESS_TOKEN'];
  final projectNumber = Platform.environment['FIREBASE_PROJECT_NUMBER'];
  final appId = Platform.environment['FIREBASE_APP_ID'];

  if (token == null || projectNumber == null || appId == null) exit(1);
  final dio = Dio();
  dio.options.headers["Authorization"] = "Bearer $token";
  final service = FirebaseCleanupService(
    apiClient: FirebaseApiClient(dio),
    projectNumber: projectNumber,
    appId: appId,
  );

  await service.executeCleanup();
}
